### PR TITLE
smoketest: Add image tag as latest for cilium agent and operator

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to add two additional flag `image.tag=latest` and
`operator.image.tag=latest` in smoketest. These two docker images are
built and loaded into cluster as part of smoke test with tag as latest.
Additionally, there is _pullPolicy_  flag set as _Never_ to make sure that 
these images will be used for testing.

Setting `image.tag=latest` and `operator.image.tag=latest` will make
sure that smoke test will work for master and any release branch v1.9.x

Signed-off-by: Tam Mach <sayboras@yahoo.com>

### CODE REVIEW

**Comment by sayboras** [comment]
Nominate this PR for backport 1.9 to resolve issue in https://github.com/cilium/cilium/pull/13664.

Backporting for 1.8 is not required due to below two reasons:
- 1.8 branch didn't have changes for recent helm refactoring
- 1.8 brach already has image tag as _latest_

**Comment by sayboras** [comment]
Full CI is not required as changes are for github action only.

**Review by rolinh — APPROVED** [review]
:100: 

Note for reviewers: this PR is required to fix smoke tests in the v1.9 branch that are currently always failing. Once merged, I'll incorporate it in the backport PR #13624 to unblock it.

**Review by aanm — APPROVED** [review]

**Pull request merged**
